term_input/src/lib.rs

```diff
@@ -180,15 +189,16 @@ impl Stream for Input {
         }
 
         // Otherwise read stdin and loop if successful
-        match self_.stdin.poll_read_ready(cx, mio::Ready::readable()) {
-            Poll::Ready(Ok(_)) => {
+        match self_.stdin.poll_read_ready(cx) {
+            Poll::Ready(Ok(mut ready)) => {
+                ready.clear_ready();
```

Comment:
I'm not sure if this is right. Perhaps we should move this to the `else` branch below.